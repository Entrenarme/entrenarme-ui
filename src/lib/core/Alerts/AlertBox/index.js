// @flow
import * as React from 'react';
import styled from 'styled-components';

import colors from '../../../helpers/colors';
import { RESOURCES_URL } from '../../../helpers/config';

import P from '../../Text/Paragraph/index';
import H5 from '../../Text/Headers/H5/index';

const setColor = (status, origin) => {
  const colorType = origin === 'icon' ? 'default' : 'light';

  switch (status) {
    case 'info':
      return colors.secondary[colorType];
    default:
      return colors.warning[colorType];
  }
};

const MainContainer = styled.div`
  display: grid;
  grid-template: ${props =>
    props.options.button ? '1fr / 50px 1fr auto' : '1fr / 50px 1fr'}
  grid-gap: 20px;
  align-items: center;
  border-radius: 4px;

  padding: 30px;
  background-color: ${props => setColor(props.options.type)};

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (max-width: 767px) {
    grid-template: ${props =>
      props.options.button
        ? 'repeat(3, auto) / auto'
        : 'repeat(2, auto) / auto'}
    justify-items: center;
    padding: 20px 15px;
  }
`;

const IconContainer = styled.div`
  display: flex;
  align-items: center;
  align-self: self-start;
  justify-content: center;
  font-size: 22px;
  width: 50px;
  height: 50px;
  mask: url(${`${RESOURCES_URL}static/images/general/hex_icon_v2.svg`});
  background-color: ${props => setColor(props.options.type, 'icon')};
  color: ${colors.white.default};
`;

const BodyContainer = styled.div`
  h5 {
    margin-bottom: 8px !important;
    color: ${props => setColor(props.options.type, 'icon')};
    font-weight: 400 !important;
    font-size: 1.125rem !important;
  }

  p {
    margin: 0px !important;
  }

  p:last-child {
    margin: 0px !important;
  }

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (max-width: 767px) {
    h5 {
      text-align: center;
      font-size: 1rem !important;
    }

    p {
      text-align: center;
    }
  }
`;

const TagsContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-top: 10px;

  > div {
    margin-top: 10px;
  }

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (max-width: 767px) {
    justify-content: center;
  }
`;

const ButtonContainer = styled.div`
  align-self: self-start;

  button {
    background-color: ${props =>
      setColor(props.options.type, 'icon')} !important;
  }

  button:hover {
    background-color: ${props =>
      setColor(props.options.type, 'icon')} !important;
    opacity: 0.9;
  }
`;

type Props = {
  icon: React.Node,
  title: React.Node,
  text: React.Node,
  tags?: Array<React.Node>,
  button?: React.Node,
  type?: 'info' | 'warning',
};

const AlertBox = ({
  icon,
  title,
  text,
  tags,
  button,
  type,
  ...rest
}: Props) => {
  return (
    <MainContainer options={{ type, button }} {...rest}>
      <IconContainer options={{ type }}>{icon}</IconContainer>
      <BodyContainer options={{ type }}>
        <H5>{title}</H5>
        <P>{text}</P>
        {tags ? <TagsContainer>{tags}</TagsContainer> : null}
      </BodyContainer>
      {button ? (
        <ButtonContainer options={{ type }}>{button}</ButtonContainer>
      ) : null}
    </MainContainer>
  );
};

AlertBox.defaultProps = {
  button: null,
  type: 'info',
};

export default AlertBox;
