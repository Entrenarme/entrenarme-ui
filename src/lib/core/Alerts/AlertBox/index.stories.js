// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAddressBook, faTimes } from '@fortawesome/pro-light-svg-icons';

import colors from '../../../helpers/colors';

import AlertBox from './index';

import Button from '../../Button/index';

import FilterTag from '../../FilterTag/index';

const title = 'title test example';

const content =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin aliquet nulla a consectetur porta. In massa lacus, condimentum a turpis at, rutrum suscipit metus. Morbi sit amet odio eget tortor bibendum pulvinar. Proin varius ac leo eu bibendum. Etiam eu lectus non nibh gravida mollis.';

const Tags = [
  <FilterTag
    title="Test"
    callback={() => console.log('hola')}
    rightIcon={<FontAwesomeIcon icon={faTimes} color={colors.text} />}
  />,
  <FilterTag
    title="Test"
    callback={() => console.log('hola')}
    rightIcon={<FontAwesomeIcon icon={faTimes} color={colors.text} />}
  />,
  <FilterTag
    title="Test"
    callback={() => console.log('hola')}
    rightIcon={<FontAwesomeIcon icon={faTimes} color={colors.text} />}
  />,
];

storiesOf('Alerts/Alert box', module).add(
  'default',
  withInfo('default')(() => (
    <AlertBox
      icon={<FontAwesomeIcon icon={faAddressBook} />}
      title={title}
      text={content}
    />
  )),
);

storiesOf('Alerts/Alert box', module).add(
  'with button',
  withInfo('with button')(() => (
    <AlertBox
      icon={<FontAwesomeIcon icon={faAddressBook} />}
      title={title}
      text={content}
      button={<Button>Test</Button>}
    />
  )),
);

storiesOf('Alerts/Alert box', module).add(
  'with button and warning',
  withInfo('with button and warning')(() => (
    <AlertBox
      icon={<FontAwesomeIcon icon={faAddressBook} />}
      title={title}
      text={content}
      button={<Button>Test</Button>}
      type="warning"
    />
  )),
);

storiesOf('Alerts/Alert box', module).add(
  'with tags',
  withInfo('with tags')(() => (
    <AlertBox
      icon={<FontAwesomeIcon icon={faAddressBook} />}
      title={title}
      text={content}
      tags={Tags}
    />
  )),
);

storiesOf('Alerts/Alert box', module).add(
  'with tags and button',
  withInfo('with tags and button')(() => (
    <AlertBox
      icon={<FontAwesomeIcon icon={faAddressBook} />}
      title={title}
      text={content}
      tags={Tags}
      button={<Button>Test</Button>}
    />
  )),
);
