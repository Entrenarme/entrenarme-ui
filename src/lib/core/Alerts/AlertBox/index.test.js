import * as React from 'react';
import { render, getByText } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAlarmClock } from '@fortawesome/free-solid-svg-icons';

import AlertBox from './index';

const title = 'title test example';

const content =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin aliquet nulla a consectetur porta. In massa lacus, condimentum a turpis at, rutrum suscipit metus. Morbi sit amet odio eget tortor bibendum pulvinar. Proin varius ac leo eu bibendum. Etiam eu lectus non nibh gravida mollis.';

const { container } = render(
  <AlertBox
    icon={<FontAwesomeIcon icon={faAlarmClock} />}
    title={title}
    text={content}
  />,
);

test('renders "AlertBox"', () => {
  expect(container.firstChild).toMatchSnapshot();
});
