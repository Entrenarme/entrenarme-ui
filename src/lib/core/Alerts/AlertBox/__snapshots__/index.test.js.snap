// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "AlertBox" 1`] = `
<div
  class="sc-htpNat jKptkh"
>
  <div
    class="sc-bxivhb hftzEY"
  />
  <div
    class="sc-ifAKCX bMRZaQ"
  >
    <h5
      class="sc-bwzfXH eeIVmL"
    >
      title test example
    </h5>
    <p
      class="sc-bdVaJa hKtTjA"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin aliquet nulla a consectetur porta. In massa lacus, condimentum a turpis at, rutrum suscipit metus. Morbi sit amet odio eget tortor bibendum pulvinar. Proin varius ac leo eu bibendum. Etiam eu lectus non nibh gravida mollis.
    </p>
  </div>
</div>
`;
